dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -2358,6 +2358,8 @@ public boolean updateAutoDoiGeneration(@Parameter(hidden = true) @Auth User user
             @RequestBody(description = "The request to update DOI generation", required = true, content = @Content(schema = @Schema(implementation = AutoDoiRequest.class))) AutoDoiRequest autoDoiRequest) {
         final Workflow workflow = workflowDAO.findById(workflowId);
         checkNotNullEntry(workflow);
+        boolean canChange = canWrite(user, workflow) || isAdmin(user) || isCurator(user);
```

Comment:
Here, we provide access to workflow owners, and maintain admin/curator access, in case we need to patch up a problem by hand.

Per our preferred practices, we are supposed to use the `@RolesAllowed` annotation to implement admin/curator access.  However, what we're trying to do here is allow access a combination of owner users and admins/curators, which our current annotation setup doesn't support.  Thus, the code here, instead.